language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/CommandUtil.java

```diff
@@ -161,10 +168,44 @@ public static Command getAllDocGenerationCommand(String docUri) {
             args.add(new CommandArgument(CommandConstants.ARG_KEY_DOC_URI, params.getTextDocument().getUri()));
             String commandTitle = CommandConstants.CREATE_FUNCTION_TITLE + " " + functionName + "(...)";
             commands.add(new Command(commandTitle, CommandConstants.CMD_CREATE_FUNCTION, args));
+        } else if (isVariableAssignmentRequired(diagnosticMessage)) {
+            BLangInvocation functionNode = getFunctionNode(params, documentManager, lsCompiler);
```

Comment:
Instead of compiling at the Code action generation phase, shall we move this (to the command execution phase? Since this is additional overhead when the user clicks on the line with the diagnostic. We need to compile and get the bLangPackage only at the execution.

Reply:
I tried to do it. Unfortunately; I could not get the cursor position for the CommandExecutor. Need to test setting cursor position at the `hover` phase and doing all calculations at the `execution` phase(note: we can pass text-only `CommandArguments`). In that case, we can move calculations of the other codeActions(i.e. importPackage, createFunction) as well. Shall we try it in a seperate phase?

Reply:
Let's convert the Range parameter of CodeActionParams to a JSON and set it as a text argument. I think that would be the ideal case to handle since we have only the text args. 
+1 for addressing this in a separate phase.